[WIP] Convert Patcher spec.json usage to temp file usage (#2546)
gruntwork-io/docs: docs/2.0/docs/patcher/guides/promotion-workflows.md

    * The `patcher update` command reads the `spec` file, checks out the repository code, commits the changes, and pushes a pull request. 
    * For the pull request workflow to function correctly, the `pull_request_branch` must follow the format `$PREFIX$DEPENDENCYID`. This format allows the workflow to track and process updates accurately. The `trigger-next-env` job strips out the prefix.

:::info
As of `v0.14.x` (`patcher-action` `v2.10.x`), Patcher has deprecated support of checking in the spec output file from a `patcher report` run into your codebase. 

> This output is intended as a temporary reference between `report` and `update` runs, and is treated as a temp file, going forward. 

Reviewer: ```suggestion
This file, similar to an OpenTofu plan file, is intended to be a temporary artifact to capture run details between `report` and `update`.   
```

> If you have this file in your codebase, it is recommended that you delete it and/or include it in the `.gitignore` file.

Reviewer: ```suggestion
We recommend that you delete and or `.gitignore` any existing spec files in your codebase.
```

Active voice + shorter.
